### PR TITLE
Add field roots for beacon state v3

### PR DESCRIPTION
**Pull request by terencechain**
Part of implementing beacon state v3 (version: Merge) pkg. This PR adds all field roots related functionalities. These files are essentially duplicated from v2

As a reviewer, please cross check they are correctly duplicated from v2

### CODE REVIEW

File: beacon-chain/state/v3/field_root_validator.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	if h.rootsCache != nil {
+		h.rootsCache.Set(string(enc), valRoot, 32)
+	}
+	return valRoot, nil
+}
+
+// ValidatorRegistryRoot computes the HashTreeRoot Merkleization of
+// a list of validator structs according to the eth2
+// Simple Serialize specification.
+func ValidatorRegistryRoot(vals []*ethpb.Validator) ([32]byte, error) {
```
Comment (potuz) [inline beacon-chain/state/v3/field_root_validator.go:84]: Since this is now exported does it need a test?

Comment (terencechain) [reply beacon-chain/state/v3/field_root_validator.go:84]: I think we can add more general tests to `v1` and `v2` as well

File: beacon-chain/state/v3/field_roots.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	// Next sync committee root.
+	nextSyncCommitteeRoot, err := stateutil.SyncCommitteeRoot(state.NextSyncCommittee)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not compute sync committee merkleization")
+	}
+	fieldRoots[23] = nextSyncCommitteeRoot[:]
+
+	// Execution payload root.
+	//TODO: Blocked by https://github.com/ferranbt/fastssz/pull/65
+	fieldRoots[24] = []byte{}
```
Comment (nisdas) [inline beacon-chain/state/v3/field_roots.go:222]: Any reason we need fast-ssz here ? All these methods use our custom ssz hashers.

**Comment by terencechain** [comment]
oops, accidentally closed this by reverting to far & force push. Will reopen

**Pull request closed without merging**